base/ca/src/main/java/com/netscape/ca/CAService.java

```diff
@@ -1725,8 +1734,10 @@ private boolean compareChallengePassword(CertRecord record, String pwd)
         return challengeString.equals(hashpwd);
     }
 
-    private String hashPassword(String pwd) {
-        String salt = "lala123";
+    private String hashPassword(String pwd) throws NotInitializedException {
+        JSSSecureRandom rand = CryptoManager.getInstance().getSecureRNG();
+        byte[] salt = new byte[32];
+        rand.nextBytes(salt);
         byte[] pwdDigest = mSHADigest.digest((salt + pwd).getBytes());
```

Comment:
I think this line should be changed like this:
```
mSHADigest.update(salt);
byte[] pwdDigest = mSHADigest.digest(pwd.getBytes());
```

Reply:
Done.